src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java

```diff
@@ -0,0 +1,132 @@
+import com.conveyal.datatools.manager.auth.Auth0Connection;
+import com.conveyal.datatools.manager.models.FeedRetrievalMethod;
+import com.conveyal.datatools.manager.models.FeedSource;
+import com.conveyal.datatools.manager.models.FetchFrequency;
+import com.conveyal.datatools.manager.models.Project;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import com.conveyal.datatools.manager.utils.HttpUtils;
+import com.conveyal.datatools.manager.utils.json.JsonUtil;
+import com.google.common.collect.ListMultimap;
+import org.apache.http.HttpResponse;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URL;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class FeedSourceControllerTest extends DatatoolsTest {
+    private static Project project = null;
+    private static FeedSource feedSourceWithUrl = null;
+    private static FeedSource feedSourceWithNoUrl = null;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        DatatoolsTest.setUp();
+        Auth0Connection.setAuthDisabled(true);
+        project = new Project();
+        project.name = "ProjectOne";
+        project.autoFetchFeeds = true;
+        Persistence.projects.create(project);
+        feedSourceWithUrl = createFeedSource("FeedSourceOne", new URL("http://www.feedsource.com"));
+        feedSourceWithNoUrl = createFeedSource("FeedSourceTwo", null);
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        Auth0Connection.setAuthDisabled(Auth0Connection.getDefaultAuthDisabled());
+        if (project != null) {
+            Persistence.projects.removeById(project.id);
+        }
+        if (feedSourceWithUrl != null) {
+            Persistence.feedSources.removeById(feedSourceWithUrl.id);
+        }
+        if (feedSourceWithNoUrl != null) {
+            Persistence.feedSources.removeById(feedSourceWithNoUrl.id);
+        }
+    }
+
+    /**
+     * Manipulate a feed source and confirm that it is correctly scheduled:
+     *  1. Create a feed source with auto fetch enabled and confirm it is scheduled.
+     *  2. Update the same feed source turning off auth fetch and confirm it is no longer scheduled.
+     *  3. Update the same feed source turning auth fetch back on and confirm it is scheduled once more.
+     */
+    @Test
+    public void createFeedSourceWithUrlTest() {
+        // create a feed source.
+        HttpResponse createFeedSourceResponse = TestUtils.makeRequest("/api/manager/secure/feedsource",
+            JsonUtil.toJson(feedSourceWithUrl),
+            HttpUtils.REQUEST_METHOD.POST
+        );
+
+        assertEquals(200, createFeedSourceResponse.getStatusLine().getStatusCode());
+
+        ListMultimap<String, ScheduledJob> scheduledJobsForFeedSources = Scheduler.scheduledJobsForFeedSources;
+        assertEquals(1, scheduledJobsForFeedSources.get(feedSourceWithUrl.id).size());
```

Comment:
I think it would be easier to read if the `scheduledJobsForFeedSources` declaration (here and below) were replaced with a method:

```java
private int jobCountForFeed(String feedSourceId) {
  return Scheduler.scheduledJobsForFeedSources.get(feedSourceId).size();
}
```

Reply:
Updated.